File: Cpp/Algorithm/src/Chapter3/PlusOne_66.cpp
#include "../Chapter3/chapter3.h"

vector<int> myAlgo::PlusOne_66::plusOne(vector<int>& digits)
{	
	vector<int> res;
	int carry = 1;
	for (int i = digits.size()-1; i >=0; i--)
	{
		res.insert(res.begin(), (digits[i] + carry) % 10);
		carry = (digits[i] + carry) / 10;
	}
	if (carry==1)
	{
		res.insert(res.begin(), 1);
	}
	return res;
}

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/TopInterview2Round/plusOne_66.java
package algo.tzashinorpu.SpecialSubject.TopInterview2Round;

public class plusOne_66 {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        int carry = 1;
        for (int i = n - 1; i >= 0; i--) {
            int sum = digits[i] + carry;
            digits[i] = sum % 10;
            carry = sum / 10;
        }
        if (carry == 1) {
            int[] ints = new int[n + 1];
            ints[0] = 1;
            System.arraycopy(digits, 0, ints, 1, n);
            return ints;
        }
        return digits;
    }
}
